packages/tracking_repository/lib/src/tracking_repository.dart

```diff
@@ -6,11 +6,11 @@ import 'package:tracking_repository/src/track_event.dart';
 class TrackingRepository {
   /// Default construcot
   TrackingRepository({
-    required this.firebase,
-  });
+    required FirebaseAnalytics firebase,
+  }) : _firebase = firebase;
 
   /// Update to configure with relevant trackers
-  final FirebaseAnalytics firebase;
+  final FirebaseAnalytics _firebase;
```

Comment:
Let's simplify this with `FirebaseAnalytics? firebase` and `_firebase = firebase ?? FirebaseAnalytics.instance`
That way you won't need to explicitly pass in the instance unless you're testing.